sdk/storage/Azure.Storage.Blobs/tests/ClientSideEncryptionTests.cs

```diff
@@ -76,32 +76,46 @@ private async Task<IKeyEncryptionKey> GetKeyvaultIKeyEncryptionKey()
         {
             if (userKeyBytes == default)
             {
                 const int keySizeBits = 256;
-                var bytes = new byte[keySizeBits >> 3];
-                new RNGCryptoServiceProvider().GetBytes(bytes);
-                userKeyBytes = bytes;
+                userKeyBytes = GetRandomBuffer(keySizeBits >> 3);
             }
-            keyId ??= Guid.NewGuid().ToString();
+            keyId ??= Recording.Random.NewGuid().ToString();
 
             var keyMock = new Mock<IKeyEncryptionKey>(MockBehavior.Strict);
             keyMock.SetupGet(k => k.KeyId).Returns(keyId);
             if (IsAsync)
             {
                 keyMock.Setup(k => k.WrapKeyAsync(s_algorithmName, IsNotNull<ReadOnlyMemory<byte>>(), s_cancellationToken))
-                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>((algorithm, key, cancellationToken) => Task.FromResult(Xor(userKeyBytes, key.ToArray())));
+                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>(async (algorithm, key, cancellationToken) =>
+                    {
+                        await Task.Delay(optionalDelay);
+                        return Xor(userKeyBytes, key.ToArray());
+                    });
                 keyMock.Setup(k => k.UnwrapKeyAsync(s_algorithmName, IsNotNull<ReadOnlyMemory<byte>>(), s_cancellationToken))
-                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>((algorithm, wrappedKey, cancellationToken) => Task.FromResult(Xor(userKeyBytes, wrappedKey.ToArray())));
+                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>(async (algorithm, wrappedKey, cancellationToken) =>
+                    {
+                        await Task.Delay(optionalDelay);
+                        return Xor(userKeyBytes, wrappedKey.ToArray());
+                    });
             }
             else
             {
                 keyMock.Setup(k => k.WrapKey(s_algorithmName, IsNotNull<ReadOnlyMemory<byte>>(), s_cancellationToken))
-                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>((algorithm, key, cancellationToken) => Xor(userKeyBytes, key.ToArray()));
+                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>((algorithm, key, cancellationToken) =>
+                    {
+                        Thread.Sleep(optionalDelay);
+                        return Xor(userKeyBytes, key.ToArray());
+                    });
                 keyMock.Setup(k => k.UnwrapKey(s_algorithmName, IsNotNull<ReadOnlyMemory<byte>>(), It.IsAny<CancellationToken>()))
-                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>((algorithm, wrappedKey, cancellationToken) => Xor(userKeyBytes, wrappedKey.ToArray()));
+                    .Returns<string, ReadOnlyMemory<byte>, CancellationToken>((algorithm, wrappedKey, cancellationToken) =>
+                    {
+                        Thread.Sleep(optionalDelay);
```

Comment:
Let's use `await Delay()` instead, so we don't actually wait in playback mode.

Reply:
We want this delay even in playback mode. A race condition is being setup in testing and if we don't pause even in playback then the requests will end up out of order.